File: Server_Spring/Elderly_Care/src/main/java/kr/spring/care/matching/dto/MatchingRequestDto.java
package kr.spring.care.matching.dto;

import lombok.Getter;
import lombok.Setter;

import jakarta.validation.constraints.Future;
import jakarta.validation.constraints.FutureOrPresent;
import jakarta.validation.constraints.NotNull;
import kr.spring.care.matching.constant.MatchingStatus;

import java.time.LocalDate;
import java.time.LocalTime;

@Getter
@Setter
public class MatchingRequestDto {

	private Long userId;
    private Long seniorId;
    private Long caregiverId;
    
    @NotNull(message = "지역 입력은 필수입니다.")
    private String matchingCountry;

    @NotNull(message = "시작 날짜는 필수입니다.")
    @FutureOrPresent(message = "시작 날짜는 현재 혹은 미래 날짜여야 합니다.")
    private LocalDate startDate;

    @NotNull(message = "종료 날짜는 필수입니다.")
    @Future(message = "종료 날짜는 미래의 날짜여야 합니다.")
    private LocalDate endDate;

    @NotNull(message = "시작 시간은 필수입니다.")    
    private LocalTime startTime;
    
    @NotNull(message = "종료 시간은 필수입니다.")
    private LocalTime endTime;

    private MatchingStatus status;
    
    private String userRole;
    
    // Senior 관련 필드
    private String health;
    private String seniorName;
    private String requirements;
    private Boolean hasGuardian;
    
    // Guardian 관련 필드
    private String guardianName;
    private String relationship;
    private String elderlyName;
    private String elderlyGender;
}

File: Client_Android/app/src/main/java/com/example/elderlycare/matching/model/MatchingRequestDto.kt
package com.example.elderlycare.matching.model

data class MatchingRequestDto(
    var userId: Long? = null,
    var caregiverId: Long? = null,
    var matchingCountry: String? = null,
    var startDate: String? = null,
    var endDate: String? = null,
    var startTime: String? = null,
    var endTime: String? = null,
    var status: String? = null,
    var userRole: String? = null,
    var health: String? = null,
    var requirements: String? = null,
    var hasGuardian: Boolean? = null,
    var guardianName: String? = null,
    var relationship: String? = null,
    var elderlyName: String? = null,
    var elderlyGender: String? = null
)